### PR TITLE
Fixed Number of Products in a WishList (if product is disabled or removed)

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Fixed Number of Products in a WishList (if product is disabled or removed)
| Type?             | bug fix
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes PrestaShop/PrestaShop#27488
| How to test?      | Cf. PrestaShop/PrestaShop#27488

### CODE REVIEW

File: classes/WishList.php [hunk 1]
```diff
@@ -269,15 +269,17 @@ public static function getAllWishlistsByIdCustomer($id_customer)
...
         } elseif (Shop::getContextShopGroupID()) {
-            $shop_restriction = 'AND id_shop_group = ' . (int) Shop::getContextShopGroupID();
+            $shop_restriction = 'AND w.id_shop_group = ' . (int) Shop::getContextShopGroupID();
         }
 
         return Db::getInstance((bool) _PS_USE_SQL_SLAVE_)->executeS('
             SELECT  w.`id_wishlist`, COUNT(wp.`id_product`) AS nbProducts, w.`name`, w.`default`, w.`token`
             FROM `' . _DB_PREFIX_ . 'wishlist_product` wp
             RIGHT JOIN `' . _DB_PREFIX_ . 'wishlist` w ON (w.`id_wishlist` = wp.`id_wishlist`)
+            RIGHT JOIN `' . _DB_PREFIX_ . 'product` p ON p.`id_product` = wp.`id_product`
```
Comment (PierreRambaud) [inline classes/WishList.php:281]:
```suggestion
            INNER JOIN `' . _DB_PREFIX_ . 'product` p ON p.`id_product` = wp.`id_product`
```

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
Juste one comment, not sure the RIGHT JOIN is the good approach

**Review by PierreRambaud — APPROVED** [review]

**Review by NeOMakinG — APPROVED** [review]

**Review by florine2623 — CHANGES_REQUESTED** [review]
Hello @Progi1984 ,
Thanks for the PR !

After installing the PR, I can't add any product to my wishlist.


https://user-images.githubusercontent.com/16019289/157653260-ae4a8efc-5fce-4345-92be-51a51dc3c197.mov


Could you check ?
Thanks!

**Comment by Progi1984** [comment]
@florine2623 Have you build assets ?

**Comment by NeOMakinG** [comment]
> Hello @Progi1984 , Thanks for the PR !
> 
> After installing the PR, I can't add any product to my wishlist.
> 
>  Screen.Recording.2022-03-10.at.12.29.44.mov 
> Could you check ? Thanks!

@florine2623 You probably need to reset the module, or test with the develop branch, this problem looks related to the core but not the module

**Review by florine2623 — APPROVED** [review]
Thanks @Progi1984 @NeOMakinG ,

Looking good to me ^^
Numbers in parenthesis match with reality !

**Comment by Progi1984** [comment]
Thanks @florine2623

**Pull request merged**